components/brave_wallet/browser/solana_instruction_builder.cc

```diff
@@ -121,15 +98,10 @@ std::optional<SolanaInstruction> TransferChecked(
   }
 
   // Instruction data is consisted of u8 instruction index and u64 amount.
-  std::vector<uint8_t> instruction_data = {
-      static_cast<uint8_t>(mojom::SolanaTokenInstruction::kTransferChecked)};
-
-  std::vector<uint8_t> amount_bytes;
-  UintToLEBytes(amount, &amount_bytes);
-  instruction_data.insert(instruction_data.end(), amount_bytes.begin(),
-                          amount_bytes.end());
-
-  instruction_data.emplace_back(decimals);
+  std::vector<uint8_t> instruction_data;
```

Comment:
either call `reserve` or create the vector with all the elements and write to it with `base::WriteSpan`.

Reply:
https://github.com/brave/brave-core/pull/26431/commits/6809115eb5276f7e5f97305fc0e0189ade29d515